data/migrations/20190117132413_recipe_ingredients.js

```diff
@@ -0,0 +1,23 @@
+
+exports.up = function(knex, Promise) {
+  return knex.schema.createTable('recipe_ingredients', tbl => {
+
+    // Rows
+
+    tbl.float('quantity').notNullable();
+
+    // Foreign Keys
+
+    tbl.integer('recipe_id').notNullable().references('id').inTable('recipes');
+    tbl.integer('ingredient_id').notNullable().references('id').inTable('ingredients');
+    tbl.integer('unit_id').references('id').inTable('unit_measurements');
```

Comment:
With how many foreign keys you have, it seems like it'd be helpful to have a function to make them. For example:
```javascript
module.exports = {
  foreign: (name, tbl=`${name}s`, isNullable=false) => {
    isNullable
      ? tbl.integer(`${name}_id`).references('id').on(tbl)
      : tbl.integer(`${name}_id`).notNullable().references('id').on(tbl); 
  }
};
```

Reply:
and then, usage:
```javascript
const { foreign } = require('./path_to/new_file');
// ...
tbl.float('quantity').notNullable();

// Foreign Keys
foreign('recipe', 'recipes', true);
foreign('ingredient', 'ingredients', true);
foreign('unit', 'unit_measurements');
```